src/rtree_intf.ml

```diff
@@ -125,8 +125,14 @@ module type S = sig
 
   val depth : t -> int
   (** [depth tree] returns the depth of the tree. *)
+
+  (* type f:f*)
+
+  val iter: tree -> (tree -> unit) -> unit
```

Comment:
```suggestion
  val iter: t -> (tree -> unit) -> unit
```

We need this to take a type `t`. This probably means you can write a function that accesses the inner tree in the implementation. Something like:

```ocaml
let rec iter' = (* your current function *)

let iter t = iter' t.tree
```

Reply:
Okay sir